### PR TITLE
heterps support pscore

### PR DESCRIPTION
**Pull request by Thunderbrook**
### PR types
New features

### PR changes
Others

### Describe
heterps支持后端ps使用pscore

### CODE REVIEW

File: paddle/fluid/distributed/service/ps_local_client.cc [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  uint32_t num_per_shard = dense_dim_per_shard(accessor->fea_dim(), 1);
+  std::vector<float> region_buffer;
+  region_buffer.resize(num_per_shard);
+  table_ptr->pull_dense(region_buffer.data(), region_buffer.size());
+
+  size_t region_idx = 0;
+  size_t region_data_idx = 0;
+  size_t shard_data_size = num_per_shard;
+  size_t shard_buffer_remain = shard_data_size * sizeof(float);
+  CHECK(shard_buffer_remain == region_buffer.size() * sizeof(float));
```
Comment (seiriosPlus) [inline paddle/fluid/distributed/service/ps_local_client.cc]: change `CHECK` to `PADDLE_ENFORCE`.

Comment (Thunderbrook) [reply paddle/fluid/distributed/service/ps_local_client.cc]: done

File: python/paddle/fluid/incubate/fleet/parameter_server/ir/trainer_pass.py [hunk 2]
```diff
@@ -159,18 +160,31 @@ def _pull_sparse_fuse(_program, pull_sparse_ops):
...
+                        outputs={"Outputs": outputs},
+                        attrs={
+                            "is_distributed": is_distributed,
+                            "padding_idx": padding_idx,
+                            "table_id": table_id,
+                            "lookup_table_version": op_type
+                        })
             else:
                 for i in range(len(inputs_idxs)):
                     distributed_idx = op_idxs[i] + 1
```
Comment (seiriosPlus) [inline python/paddle/fluid/incubate/fleet/parameter_server/ir/trainer_pass.py:190]: else也需要加入相同的逻辑.

**Review by seiriosPlus — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
LGTM for WITH_HETERPS

**Review by fuyinno4 — APPROVED** [review]

**Review by raindrops2sea — APPROVED** [review]

**Pull request merged**